### DISCUSSION TITLE
use pandoc from a docker container

### DISCUSSION THREAD

**Discussion by luerhard** [Zettlr/Zettlr#3532] [General]
Hi there,
First of all, many thanks for this great tool for academic writing!

I really love it but have some difficulties with cooperating with colleagues. We use quite a lot of custom-written plugins for pandoc and therefore need pandoc plugins like [sergiocorreia](https://github.com/sergiocorreia) / [panflute](https://github.com/sergiocorreia/panflute) and [ickc](https://github.com/ickc) / [pantable](https://github.com/ickc/pantable) installed (for our template see [luerhard](https://github.com/luerhard) / [academic-pandoc](https://github.com/luerhard/academic-pandoc)).

The current way of ensuring that everyone has a working, identical pandoc version with all necessary plugins is to use a docker container with everything preinstalled.

**my question**:
Is there any way to invoke the docker container from inside Zettlr and use its version of pandoc?

**Comment by nathanlesage** [comment]
Not directly, but I may have a sneaky idea: if you can make it so that you invoke the docker container like pandoc, i.e. by typing pandoc in the command line and providing arguments, then all it should take is to uncheck the setting to use the internal pandoc.

Zettlr checks at start whether there is a command called `pandoc` on the system. And if it is, it will just call that with the regular pandoc command line arguments.

If you have to perform some magic before actually invoking the container, you could write a small shell script and register that as the pandoc command on the system, take the CLI arguments and pass them on to the docker container!

Just remember that for best results, the shell script should keep running until the docker container is finished so that Zettlr can then automatically open the corresponding exported file!

**Reply by luerhard** [reply]
That indeed sounds sneaky :) Thanks for the answer!

But it also sounds like configuring everything to work around Zettlr. Tbh, I was hoping for a hidden setting to configure the pandoc path. It feels very odd to me that Zettlr allows the execution of an external pandoc executable but does not allow specifying a path to said executable.

**Reply by nathanlesage** [reply]
Well, we had that in earlier versions, but it proved to be much more difficult for users to do than we expected. Feel free to open an issue for that, though.
